fMRIVolume/GenericfMRIVolumeProcessingPipeline.sh

```diff
@@ -220,6 +220,43 @@ then
 	log_Err_Abort "the --usejacobian option must be 'true' or 'false'"
 fi
 
+
+# ------------------------------------------------------------------------------
+#  Legacy Style Data Options
+# ------------------------------------------------------------------------------
+
+DoSliceTimeCorrection=`opts_GetOpt1 "--doslicetime" $@`              # Whether to do slicetime correction (TRUE), FALSE to omit
+SliceCorrectionDirection=`opts_GetOpt1 "--slicetimedir" $@`          # direction: empty for up, --down for down
+SliceCorrectionInterleaved=`opts_GetOpt1 "--slicetimeodd" $@`        # interleaved: empty for not, --odd for interleaved
```

Comment:
There are more details than interleaving or not, and "not odd" doesn't mean "not interleaved".  The simple options act sort of like TV interlacing: progressive scan (1, 2, 3, 4...) or 2 interleaved passes, which can start with odd (1, 3, 5, 7...2, 4, 6...), or start with even (2, 4, 6...1, 3, 5...).

If the scanner sequence and the nifti orientation are different, though, you can get weirder things like reversed order (131, 129, 127...3, 1, 130, 128...), or the slice direction could be the first or second dimension, rather than the third (this is more likely in NHP scanned chest-down, chin-up or such).

Even more involved are multiband acquisitions, where you have sets of widely-spaced slices acquired at the same time, and I have no idea what the order of the sets usually is, the simplest case would be something like ((1 31 61 91), (2 32 62 92)...).

Obviously, the options don't need to support all this (I imagine the tool used doesn't support particularly many variations), but we should at least be careful with our option names, to make sure we are using the correct terminology, so that we can add additional options later if they become necessary.

Reply:
In our experience with legacy data, direction (up or down) and interleaved or not covered the situations we ran into. Regarding "odd", the parameter refers directly to FSL `slicetimer`:

```
--odd	use interleaved acquisition
```

I agree that it does not cover all possibilities, but we wanted to match with the tool used. The extended option that the `slicetimer` does offer is to provide a file with explicit slice timing information. This would expand the options significantly, but would require also additional coding. I'd suggest we mark it down as future development.

Reply:
In this case, I would say that just because FSL's option names are bad, doesn't mean ours should be bad.

As I said, I am not suggesting that we implement support for anything more complicated now, I was explaining why using terms like "odd" when we mean "interleave" is a bad idea for future work.

Reply:
Agreed. I changed the parameter name to `--interleavedslices` with options TRUE/FALSE, TRUE being the default.

Reply:
The default with the previous option name appeared to be non-interleaved, while now the default is interleaved.  I would generally expect "not specified" to mean "false" for this type of option naming, so this is looks a little confusing to me.  Do you have a reason for interleaving to be the default?

Reply:
Overall interleaved acquisition is far more common if not almost exclusive, that's the reason I think it better should be default.

Reply:
How does this tool account/allow for the fact that Siemens interleaved acquisitions in the context of an even number of slices start with the _second_ slice (counting from 1)?  That's not your problem per se, but we do want to try to help users avoid doing something completely wrong.

Reply:
My suggestion at this point would be to add a warning and then in the next iteration offer possibility of using a slice timing file.

Reply:
Sure, if the number of slices is even, and interleaved is selected, we should probably have a `log_Warn` message about differences between vendors as to how the data is acquired.  But to craft the warning in an informative way, we need to know what `slicetimer` assumes as its default.  (After all, since Oxford is a Siemens shop, they may have written `slicetimer` to work correctly with the goofy Siemens convention).  Do you know what `slicetimer` is assuming in that regard?  We should also add some in-script comments on this particular issue.

Reply:
@mharms one other option is to ask for all the parameters to be passed to  `slicetimer` in a comma or pipe separted script e.g.: "--down|--odd|-d|2", which we split up and pass to `slicetimer`. The users then have all the `slicetimer` options available and can provide them in a single parameter. They just need to look up the `slicetimer` manual.

Reply:
In Qu|Nex we mostly use either `|` or `,`. And I'd prefer staying with those. If comma is the intersection then let's go with that.

Reply:
Well, to my knowledge, we don't use either `|` or `,`, so there really isn't any "intersection".  I'd wait for @coalsont to chime in on this one before you do anything.

Reply:
@coalsont would comma or pipe be acceptable to you?

Reply:
The pipelines have only used @ for lists with only one delimiter, so there is definitely an established convention, and breaking it would look inconsistent.  There is a trick to splitting on a delimiter with space-safety, it works like this:

https://github.com/Washington-University/HCPpipelines/blob/5d82838467c2e4a1183598b09d2b00fcc7c8ffbf/global/scripts/RSNregression.sh#L104

That said, the use of @ was started by Matt, and I don't have any real attachment to any specific character.

If we wanted things to be simple for users, at the cost of space-safety (for filename options), we could have the arguments separated by spaces instead (so the user would need to use quotes), and then not quote the expansion of the variable in the slicetimer call.  This would need some testing and possible edits to make sure the quoted spaces make it through the existing argument parsing code intact.

Reply:
To be clear: I feel that we should use @ if we use a delimiter, for consistency.  There is no inherent reason for using @ rather than another character, it is merely that it is what we use everywhere else.

Reply:
We never use `@` in  Qu|Nex, but we already do parsing in Qu|Nex to create `@` separated strings, so a parameter to Qu|Nex can conform to Qu|Nex convention and we then convert it to HCP pipelines convention when passed to HCP pipelines. Unfortunately, there are too many intermediate steps for a number of reasons already that we can not use space as a delimiter.